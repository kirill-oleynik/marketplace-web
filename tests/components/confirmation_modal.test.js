const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const { ConfirmationModal } = require('../../app/components/confirmation_modal');

const close = jest.fn();
const action = jest.fn();

const renderConfirmationModal = () => (
  <ConfirmationModal
    isOpen
    close={close}
    action={action}
    t={(translation) => translation}
  />
);

describe('#render', () => {
  test('it renders correctly', () => {
    const component = shallow(
      renderConfirmationModal()
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#handleAcceptClick', () => {
  test('it calls close and action', () => {
    const component = shallow(
      renderConfirmationModal()
    );

    component.instance().handleAcceptClick();

    expect(close).toHaveBeenCalled();
    expect(action).toHaveBeenCalled();
  });
});
