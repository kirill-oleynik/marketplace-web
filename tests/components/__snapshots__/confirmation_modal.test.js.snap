// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransitionTimeout={150}
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransitionTimeout={300}
  toggle={[Function]}
  zIndex={1050}
>
  <ModalHeader
    tag="h4"
    toggle={[Function]}
    wrapTag="div"
  />
  <ModalBody
    tag="div"
  >
    confirmation.text
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      confirmation.accept
    </Button>
    <Button
      color="secondary"
      onClick={[Function]}
      tag="button"
    >
      confirmation.cancel
    </Button>
  </ModalFooter>
</Modal>
`;
